core/templates/dev/head/domain/collection/CollectionRightsBackendApiServiceSpec.js

```diff
@@ -83,9 +79,7 @@ describe('Collection rights backend API service', function() {
     var failHandler = jasmine.createSpy('fail');
 
     $httpBackend.expect(
-      'PUT', '/collection_editor_handler/rights/0').respond(200, {
-        is_private: true
-      });
+      'PUT', '/collection_editor_handler/publish/0').respond(200);
```

Comment:
You'll need to check what setCollectionPublic() below actually does; consider adding console.log() statements to your local Karma tests to debug what's going on and what handlers are being called. (This is why you need to get the Karma tests working on your local machine.)